### ISSUE TITLE
Revoke Invite does not actually revoke invite

### ISSUE THREAD

**Issue by Pyroboy5000** [Yellow-Dog-Man/Resonite-Issues#1101]
### Describe the bug?

revoke invite does not disallow said user to join via join button on previous invite. this is an issue because sessions are hidden for certain content, and users who may be offended would be able to rejoin after being revoked.

### To Reproduce

revoking invite in a hidden world still allows user to join, only in a private world does it seem to not allow a rejoin. 

### Expected behavior

Revoke invite should break previous invite so that the join button no longer works and hidden worlds cant be rejoined from a revoked invite

### Resonite Version Number

current

### What Platforms does this occur on?

Windows

### Log Files

na

### Reporters

Dante, Veer

**Comment by Banane9** [comment]
I'd wager that this is, because all you need to join a hidden world that is otherwise accessible to you (e.g. contacts(+) or open) is to know it exists. So there is no cryptographic invite key to revoke access to.

**Comment by Pyroboy5000** [comment]
It's as you expect shifty. Dante was saying basically the same thing, and my expectation was for it to break the join button on the invite so that it can't be reached again. Yes people could go around that for a hidden contacts+ session, but it would mean that functionally someone would have to be actively seeking to break back into the session. 

I placed it in bug reports based off my expectations of the feature, but it could also possibly be considered a feature change/request?

**Comment by Banane9** [comment]
> but in future, it is probably something that could be better communicated to the user.

Wouldn't it make sense to keep this open until that is the case, @shiftyscales?

Even just removing the option for "revoke invite" like it is for other public sessions types at the moment would prevent misunderstandings like this, by removing the point of confusion entirely.

**Comment by Pyroboy5000** [comment]
Veer asked me to put in the ticket due to the confusing nature it is to a user. i think it should at least be reviewed by the team.

**Comment by Pyroboy5000** [comment]
it should ideally work for hidden worlds though. i dont want an incident with a minor accessing inappropriate content if i revoke their invite

**Comment by Pyroboy5000** [comment]
Right. I'm just trying to give an example how it could be a bad thing. If I thought it worked that way I'm sure others would think so too. If the host "revokes" an invite they sent, they would assume that person can't return unless another person sent an invite. Yes other control methods can be used, but this not functioning as I described it should might lead to such incidents where someone doesn't know how it functions

**Comment by Banane9** [comment]
> As noted above- Contacts+, and Anyone sessions, even with 'don't show in session lists' are still public sessions, and would not meet the above criteria for session access.

> If I thought it worked that way I'm sure others would think so too. If the host "revokes" an invite they sent, they would assume that person can't return unless another person sent an invite. Yes other control methods can be used, but this not functioning as I described it should might lead to such incidents where someone doesn't know how it functions

I would agree with Pyro here; I suspect even many seasoned veterans wouldn't be certain that hidden worlds *aren't* enough to satisfy the guidelines on adult content. I've certainly seen at least semi-nsfw sessions as hidden Contacts+.
